ocaml/idl/datamodel_host.ml

```diff
@@ -1668,6 +1668,35 @@ let cleanup_pool_secret =
       ]
     ~allowed_roles:_R_LOCAL_ROOT_ONLY ~hide_from_docs:true ()
 
+let host_numa_affinity_policy =
+  Enum
+    ( "host_numa_affinity_policy"
+    , [
+        ("any", "VMs are spread across all available NUMA nodes")
+      ; ( "best-effort"
+        , "VMs are placed on the smallest number of NUMA nodes that they fit \
+           using soft-pinning"
+        )
+      ; ( "default"
+        , "Use the NUMA affinity policy that is the default for the current \
+           version"
+        )
+      ]
+    )
+
+let set_numa_affinity_policy =
+  call ~name:"set_numa_affinity_policy" ~lifecycle:[]
+    ~doc:"Set VM placement NUMA affinity policy"
+    ~params:
+      [
+        (Ref _host, "self", "The host")
+      ; ( host_numa_affinity_policy
+        , "value"
+        , "The NUMA affinity policy to apply to a host"
+        )
+      ]
+    ~allowed_roles:_R_LOCAL_ROOT_ONLY ()
```

Comment:
Really local root? Not pool admin?

Reply:
Ah initialy I wanted this to be a pool level setting, with the host level setting internal. But since there is no pool level setting I can change this back.